chalk-ir/src/fold.rs

```diff
@@ -527,17 +642,17 @@ where
     }
 }
 
-impl<C: Context, TF: TypeFamily> Fold<TF> for ExClause<C>
+impl<C: Context, TF: TypeFamily> Fold<TF, TF> for ExClause<C>
```

Comment:
Why isn't `Fold<TF, TTF>` implemented for `ExClause`, `FlounderedSubgoal` and `Literal`? Are they never useful?

Reply:
Yeah, just not needed. I could probably generalize it anyway, I didn't want to mess with that stuff.